qiskit/qpy/__init__.py

```diff
@@ -126,6 +126,46 @@
 by ``num_circuits`` in the file header). There is no padding between the
 circuits in the data.
 
+.. _qpy_version_7:
+
+Version 7
+=========
+
+Version 7 adds support for :class:`.~Reference` instruction and serialization of
+a :class:`.~ScheduleBlock` program while keeping its reference to subroutines::
+
+    from qiskit import pulse
+    from qiskit import qpy
+
+    with pulse.build() as schedule:
+        pulse.reference("cr45p", "q0", "q1")
+        pulse.reference("x", "q0")
+        pulse.reference("cr45p", "q0", "q1")
+
+    with open('template_ecr.qpy', 'wb') as fd:
+        qpy.dump(schedule, fd)
+
+Conventional :ref:`qpy_schedule_block` data model is preserved,
+but it is immediately followed by an extra :ref:`qpy_mapping` utf8 bytes block
+representing the data of the referenced subroutines.
```

Comment:
We probably should also document that the `SCHEDULE_BLOCK_HEADER` now has a `num_references` field in it too?

Reply:
Thanks for catching this. I forgot to remove `SCHEDULE_BLOCK_HEADER_v7` (I was thinking different data model in the beginning). Since we already have [mapping data model](https://github.com/Qiskit/qiskit-terra/blob/5672c70649b4f83a6c7146c68a6a543d4a3908fd/qiskit/qpy/formats.py#L261-L263) in QPY (key value must be string), I just reused it to express the reference mapping. Namely schedule block and references are stored back to back but they are different data section.